<a href="https://colab.research.google.com/github/manudsanun/python_dataviz2024/blob/main/Immuno_oncology_Gene_Panel__Gene_List_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [1]:
import pandas as pd
from google.colab import drive

In [2]:
drive.mount('/content/drive')

Mounted at /content/drive


# Vizgen

## นำเข้าข้อมูลยีนจาก NCBI


In [13]:
# Melanoma[All Fields] AND "Homo sapiens"[porgn] AND ("matches ensembl"[Properties] AND alive[prop] เลือกเฉพาะข้อมูลของ Human
ncbi = pd.read_csv('/content/drive/MyDrive/Spatial/Melanoma_human.txt', delimiter='\t')
ncbi

tax_id      Org_name     GeneID  CurrentID Status     Symbol  \
0       9606  Homo sapiens       1029          0   live     CDKN2A   
1       9606  Homo sapiens       1026          0   live     CDKN1A   
2       9606  Homo sapiens       7015          0   live       TERT   
3       9606  Homo sapiens       4157          0   live       MC1R   
4       9606  Homo sapiens      64135          0   live      IFIH1   
...      ...           ...        ...        ...    ...        ...   
2337    9606  Homo sapiens  100616175          0   live    MIR4633   
2338    9606  Homo sapiens     158035          0   live     REXO6P   
2339    9606  Homo sapiens  122965330          0   live    IGSF3P1   
2340    9606  Homo sapiens  100847051          0   live    MIR5194   
2341    9606  Homo sapiens  101927559          0   live  CEBPB-AS1   

                                                Aliases  \
0     ARF, CAI2, CDK4I, CDKN2, CMM2, INK4, INK4A, ML...   
1     CAP20, CDKN1, CIP1, MDA-6, P21, SDI1, WAF1, p2...   
2     CMM9, DKCA2, DKCB4, EST2, PFBMFT1, TCS1, TP2, ...   
3                                    CMM5, MSH-R, SHEP2   
4     AGS7, Hlcd, IDDM19, IMD95, MDA-5, MDA5, RLR-2,...   
...                                                 ...   
2337                                                NaN   
2338                     C9orf14, LINC00032, NCRNA00032   
2339                                                NaN   
2340                                           mir-5194   
2341                                                NaN   

                                      description  \
0            cyclin dependent kinase inhibitor 2A   
1            cyclin dependent kinase inhibitor 1A   
2                telomerase reverse transcriptase   
3                         melanocortin 1 receptor   
4     interferon induced with helicase C domain 1   
...                                           ...   
2337                                microRNA 4633   
2338                RNA exonuclease 6, pseudogene   
2339                           IGSF3 pseudogene 1   
2340                                microRNA 5194   
2341                        CEBPB antisense RNA 1   

                                     other_designations map_location  \
0     cyclin-dependent kinase inhibitor 2A|CDK4 inhi...       9p21.3   
1     cyclin-dependent kinase inhibitor 1|CDK-intera...       6p21.2   
2     telomerase reverse transcriptase|telomerase ca...      5p15.33   
3     melanocyte-stimulating hormone receptor|MC1-R|...      16q24.3   
4     interferon-induced helicase C domain-containin...       2q24.2   
...                                                 ...          ...   
2337                                       hsa-mir-4633       5q23.3   
2338          long intergenic non-protein coding RNA 32       9p21.2   
2339                                                NaN        13q11   
2340                                       hsa-mir-5194      8q24.21   
2341                                                NaN     20q13.13   

     chromosome genomic_nucleotide_accession.version  \
0             9                         NC_000009.12   
1             6                         NC_000006.12   
2             5                         NC_000005.10   
3            16                         NC_000016.10   
4             2                         NC_000002.12   
...         ...                                  ...   
2337          5                         NC_000005.10   
2338          9                         NC_000009.12   
2339         13                         NC_000013.11   
2340          8                         NC_000008.11   
2341         20                         NC_000020.11   

      start_position_on_the_genomic_accession  \
0                                    21967752   
1                                    36676463   
2                                     1253167   
3                                    89918862   
4                                   

1. "matches ensembl"[Properties]
เป็นการค้นหาข้อมูลใน NCBI ที่มีการจับคู่ (matches) หรือสอดคล้องกับข้อมูลจาก Ensembl database.
ใช้เพื่อกรองเฉพาะรายการที่มีการอ้างอิงหรือเชื่อมโยงกับ transcript, gene, หรือ protein ID จาก Ensembl.

2. alive[prop]
เป็นการกรองข้อมูลเพื่อแสดงเฉพาะรายการที่ยัง active หรือมีสถานะใช้งาน (alive) ในระบบฐานข้อมูล.
ช่วยตัดข้อมูลที่ถูกระงับหรือเลิกใช้งานออกไป.

## นำเข้าข้อมูล Immuno-oncology Gene Panel

In [14]:
panel = pd.read_csv('/content/drive/MyDrive/Spatial/Gene Panel _Gene List.csv', encoding='latin-1')
# or 'cp1252' or any other encoding that might work for your data
panel.head()

Gene                                             Classes  \
0    ACKR3  Atypical chemokine receptors;Cytokine Receptor...   
1    ACTA2                      Cancer-associated fibroblasts   
2  ADAMTS4                                  Matrix remodeling   
3     AKT1                                   Cancer Signaling   
4     AKT2                                   Cancer Signaling   

                                         Description               ID  \
0                      atypical chemokine receptor 3  ENST00000272928   
1  Myofibroblasts acquire contractile stress fibe...  ENST00000224784   
2  ADAMTSs are a family of secreted proteinases t...  ENST00000367996   
3                      AKT serine/threonine kinase 1  ENST00000407796   
4                      AKT serine/threonine kinase 2  ENST00000392038   

      HGNC ID Location  Unnamed: 6  Unnamed: 7  
0  HGNC:23692   2q37.3         NaN         NaN  
1         NaN      NaN         NaN         NaN  
2         NaN      NaN         NaN         NaN  
3    HGNC:391      NaN         NaN         NaN  
4    HGNC:392  19q13.2         NaN         NaN

In [15]:
# เปลี่ยนชื่อคอลัมน์ของ panel จาก "Gene" เป็น "Symbol"
panel = panel.rename(columns={"Gene ": "Symbol"})
panel

Symbol                                            Classes  \
0      ACKR3  Atypical chemokine receptors;Cytokine Receptor...   
1      ACTA2                      Cancer-associated fibroblasts   
2    ADAMTS4                                  Matrix remodeling   
3       AKT1                                   Cancer Signaling   
4       AKT2                                   Cancer Signaling   
..       ...                                                ...   
495     XCR1                          Macrophage and DC traffic   
496     YAP1                                      Hippo pathway   
497    ZAP70                                     Effector cells   
498    ZBED2   Transcriptional regulator and competes iwht IRF1   
499     ZEB1       Transcriptional repressor in immune response   

                                           Description               ID  \
0                        atypical chemokine receptor 3  ENST00000272928   
1    Myofibroblasts acquire contractile stress fibe...  ENST00000224784   
2    ADAMTSs are a family of secreted proteinases t...  ENST00000367996   
3                        AKT serine/threonine kinase 1  ENST00000407796   
4                        AKT serine/threonine kinase 2  ENST00000392038   
..                                                 ...              ...   
495  With the recognition of cross-presenting DCs a...  ENST00000309285   
496          Yes1 associated transcriptional regulator  ENST00000282441   
497  ZAP-70 (?-associated protein of 70 kDa) is a t...  ENST00000463643   
498                  zinc finger BED-type containing 2  ENST00000317012   
499               zinc finger E-box binding homeobox 1  ENST00000320985   

        HGNC ID  Location  Unnamed: 6  Unnamed: 7  
0    HGNC:23692    2q37.3         NaN         NaN  
1           NaN       NaN         NaN         NaN  
2           NaN       NaN         NaN         NaN  
3      HGNC:391       NaN         NaN         NaN  
4      HGNC:392   19q13.2         NaN         NaN  
..          ...       ...         ...         ...  
495         NaN       NaN         NaN         NaN  
496  HGNC:16262   11q22.1         NaN         NaN  
497         NaN       NaN         NaN         NaN  
498  HGNC:20710   3q13.13         NaN         NaN  
499  HGNC:11642  10p11.22         NaN         NaN  

[500 rows x 8 columns]

In [16]:
panel1 = panel['Symbol']
panel1

0        ACKR3
1        ACTA2
2      ADAMTS4
3         AKT1
4         AKT2
        ...   
495       XCR1
496       YAP1
497      ZAP70
498      ZBED2
499       ZEB1
Name: Symbol, Length: 500, dtype: object

## Melanoma : NCBI เทียบกับ GeneCard

In [5]:
GeneCard = pd.read_csv('/content/drive/MyDrive/Spatial/GeneCards-SearchResults.csv')
GeneCard

Gene Symbol                                        Description  \
0            CDKN2A               Cyclin Dependent Kinase Inhibitor 2A   
1              CDK4                          Cyclin Dependent Kinase 4   
2              MC1R                            Melanocortin 1 Receptor   
3              MITF           Melanocyte Inducing Transcription Factor   
4              BRAF      B-Raf Proto-Oncogene, Serine/Threonine Kinase   
...             ...                                                ...   
11068  LOC111089942                              Pleiotrophin Promoter   
11069  LOC117038771  CRISPRi-FlowFISH-Validated H1-10 Regulatory El...   
11070  LOC129937550  ATAC-STARR-Seq Lymphoblastoid Active Region 20512   
11071        ORI343                          Origin Of Replication 343   
11072         SSXP8                            SSX Family Pseudogene 8   

                 Category Uniprot ID  Gifts        GC Id  Relevance score  
0          Protein Coding     Q8N726     63  GC09M021967       146.845993  
1          Protein Coding     P11802     66  GC12M059932       116.223862  
2          Protein Coding     Q01726     59  GC16P089912        91.914803  
3          Protein Coding     O75030     61  GC03P069740        90.593117  
4          Protein Coding     P15056     66  GC07M140783        87.288742  
...                   ...        ...    ...          ...              ...  
11068  Functional Element        NaN      2  GC07P137343         0.101479  
11069  Functional Element        NaN      2  GC03P128487         0.101479  
11070  Functional Element        NaN      2  GC03P141520         0.101479  
11071  Functional Element        NaN      2  GC06P156840         0.101479  
11072          Pseudogene        NaN      2  GC0XP048276         0.101479  

[11073 rows x 7 columns]

ใน GeneCard มีข้อมูลทั้งหมด 11073 ยีน

### เปรียบเทียบข้อมูล Melanom gene จาก NCBI กับ GeneCard

In [10]:
# แสดงเฉพาะคอลัมน์ Gene
GeneCard1 = GeneCard['Gene Symbol']

# เปลี่ยนชื่อคอลัมน์ให้เหมือนกันเพื่อความสะดวกในการเปรียบเทียบ
GeneCard1 = GeneCard1.rename('Symbol')

# รวมตารางเข้าด้วยกันโดยใช้คอลัมน์ 'Symbol' เป็นตัวเชื่อม
NCBI_GeneCard = pd.merge(ncbi, GeneCard1, on='Symbol', how='inner')

# นับจำนวนแถวในตารางที่รวมกันแล้ว (ซึ่งก็คือจำนวนยีนที่ตรงกัน)
matched_genes_count = len(NCBI_GeneCard)

# แสดงผลลัพธ์
print(f"จำนวนยีนที่ตรงกันระหว่างข้อมูลจาก GeneCard และ NCBI คือ {matched_genes_count} ยีน")

# แสดงตารางที่รวมกันแล้ว
NCBI_GeneCard1 = NCBI_GeneCard['Symbol']
NCBI_GeneCard1

จำนวนยีนที่ตรงกันระหว่างข้อมูลจาก GeneCard และ NCBI คือ 2335 ยีน


0          CDKN2A
1          CDKN1A
2            TERT
3            MC1R
4           IFIH1
          ...    
2330      MIR4633
2331       REXO6P
2332      IGSF3P1
2333      MIR5194
2334    CEBPB-AS1
Name: Symbol, Length: 2335, dtype: object

จำนวนยีนที่ตรงกันระหว่างข้อมูลจาก GeneCard และ NCBI คือ 2335 ยีน

### เปรียบเทียบยีน Melanoma 2335 ยีนที่ได้จากการ match ระหว่าง GeneCard และ NCBI กับ panel ของ Vizgen

In [17]:
# รวมตารางเข้าด้วยกันโดยใช้คอลัมน์ 'Symbol' เป็นตัวเชื่อม
melanoma_vizgen = pd.merge(NCBI_GeneCard1, panel1, on='Symbol', how='inner')

# นับจำนวนแถวในตารางที่รวมกันแล้ว (ซึ่งก็คือจำนวนยีนที่ตรงกัน)
matched_genes = len(melanoma_vizgen)

# แสดงผลลัพธ์
print(f"จำนวนยีนที่ match กันระหว่างข้อมูลจาก panel และ NCBI_GeneCard คือ {matched_genes} ยีน")

# แสดงตารางที่รวมกันแล้ว (ถ้าต้องการ)
melanoma_vizgen

จำนวนยีนที่ match กันระหว่างข้อมูลจาก panel และ NCBI_GeneCard คือ 299 ยีน


Symbol
0    CDKN1A
1     CXCL1
2      BRAF
3       FAP
4      TP53
..      ...
294    IDO2
295   IL4I1
296   PREX2
297  CLEC4C
298   CCR10

[299 rows x 1 columns]

In [18]:
# prompt: ลบข้อมูลที่ซ้ำในตาราง melanoma_vizgen ออก

# Drop duplicate rows based on the 'Symbol' column
melanoma_vizgen = melanoma_vizgen.drop_duplicates(subset='Symbol')

# Display the DataFrame after removing duplicates
print(f"จำนวนยีนที่ตรงกันหลังจากลบข้อมูลซ้ำ: {len(melanoma_vizgen)} ยีน")
melanoma_vizgen

จำนวนยีนที่ตรงกันหลังจากลบข้อมูลซ้ำ: 299 ยีน


Symbol
0    CDKN1A
1     CXCL1
2      BRAF
3       FAP
4      TP53
..      ...
294    IDO2
295   IL4I1
296   PREX2
297  CLEC4C
298   CCR10

[299 rows x 1 columns]

จำนวนยีนที่ตรงกันระหว่างข้อมูลจาก panel และ NCBI_GeneCard คือ 299 ยีน

In [19]:
# prompt: นำข้อมูลออกเป็นไฟล์ excel
# Specify the output file path in your Google Drive, including the file extension (.xlsx)
output = '/content/drive/MyDrive/Spatial/melanoma_vizgen.xlsx'  # Changed this to include .xlsx

# Export the merged DataFrame to an Excel file
melanoma_vizgen.to_excel(output, index=False)  # index=False prevents writing row indices to the file

print(f"Merged data exported to: {output}")

Merged data exported to: /content/drive/MyDrive/Spatial/melanoma_vizgen.xlsx


## Hepatocellular carcinoma

1. นำเข้าข้อมูล Hepatocellular carcinoma genes จาก NCBI เอาเฉพาะยีนใน Human
2. เอา HCC genes ไปเทียบกับ Immuno-oncology Gene Panel _Gene List ทั้ง 500 gene

In [20]:
# นำเข้าข้อมูลยีน Hepatocellular carcinoma ลองเทียบกับยีนทั้งหมดที่มีในตับดู
HCC= pd.read_csv('/content/drive/MyDrive/Spatial/Hepatocellular_carcinoma_human.txt', delimiter='\t')
HCC

tax_id      Org_name     GeneID  CurrentID Status      Symbol  \
0       9606  Homo sapiens       7157          0   live        TP53   
1       9606  Homo sapiens       1499          0   live      CTNNB1   
2       9606  Homo sapiens      22974          0   live        TPX2   
3       9606  Homo sapiens      55908          0   live     ANGPTL8   
4       9606  Homo sapiens       3123          0   live    HLA-DRB1   
...      ...           ...        ...        ...    ...         ...   
4369    9606  Homo sapiens  101927459          0   live   LINC01608   
4370    9606  Homo sapiens     414761          0   live       HCG15   
4371    9606  Homo sapiens  100507300          0   live  ALKBH3-AS1   
4372    9606  Homo sapiens  100506033          0   live   PTOV1-AS1   
4373    9606  Homo sapiens  116435281          0   live       EGILA   

                                                Aliases  \
0                         BCC7, BMFS5, LFS1, P53, TRP53   
1                 CTNNB, EVR7, MRD19, NEDSDV, armadillo   
2     C20orf1, C20orf2, DIL-2, DIL2, FLS353, GD:C20o...   
3                C19orf80, PRO1185, PVPA599, RIFL, TD26   
4                          DRB1, HLA-DR1B, HLA-DRB, SS1   
...                                                 ...   
4369                                                NaN   
4370                                              HCG16   
4371                                                NaN   
4372                                                NaN   
4373                                           Lnc-EGFR   

                                            description  \
0                                     tumor protein p53   
1                                        catenin beta 1   
2                    TPX2 microtubule nucleation factor   
3                                   angiopoietin like 8   
4     major histocompatibility complex, class II, DR...   
...                                                 ...   
4369        long intergenic non-protein coding RNA 1608   
4370                               HLA complex group 15   
4371                             ALKBH3 antisense RNA 1   
4372                              PTOV1 antisense RNA 1   
4373                            EGFR interacting lncRNA   

                                     other_designations map_location  \
0     cellular tumor antigen p53|antigen NY-CO-13|mu...      17p13.1   
1     catenin beta-1|catenin (cadherin-associated pr...       3p22.1   
2     targeting protein for Xklp2|TPX2, microtubule-...     20q11.21   
3     angiopoietin-like protein 8|hepatocellular car...      19p13.2   
4     major histocompatibility complex, class II, DR...      6p21.32   
...                                                 ...          ...   
4369                                     TCONS_00015136       8q23.2   
4370  HLA complex group 15 (non-protein coding)|HLA ...       6p22.1   
4371                                                NaN      11p11.2   
4372         PTOV1 antisense RNA 1 (non-protein coding)     19q13.33   
4373                                                NaN      14q11.2   

     chromosome genomic_nucleotide_accession.version  \
0            17                         NC_000017.11   
1             3                         NC_000003.12   
2            20                         NC_000020.11   
3            19                         NC_000019.10   
4             6                         NC_000006.12   
...         ...                                  ...   
4369          8                         NC_000008.11   
4370          6                         NC_000006.12   
4371         11                         NC_000011.10   
4372         19                         NC_000019.10   
4373         14                          NC_000014.9   

      start_position_on_the_genomic_accession  \
0                                     7668421   
1                                    41199505   
2                                    31739290   
3

In [21]:
HCC1 = HCC['Symbol']
HCC1

0             TP53
1           CTNNB1
2             TPX2
3          ANGPTL8
4         HLA-DRB1
           ...    
4369     LINC01608
4370         HCG15
4371    ALKBH3-AS1
4372     PTOV1-AS1
4373         EGILA
Name: Symbol, Length: 4374, dtype: object

### เปรียบเทียบข้อมูล HCC gene จาก NCBI กับ GeneCard

In [22]:
GeneCard_HCC = pd.read_csv('/content/drive/MyDrive/Spatial/GeneCards_Hepatocellular carcinoma.csv')
GeneCard_HCC

Gene Symbol                                   Description  \
0                 TP53                             Tumor Protein P53   
1                  MET  MET Proto-Oncogene, Receptor Tyrosine Kinase   
2                BRCA1                   BRCA1 DNA Repair Associated   
3                  ATM                   ATM Serine/Threonine Kinase   
4                 MSH6                                MutS Homolog 6   
...                ...                                           ...   
11933          MIR6881                                 MicroRNA 6881   
11934  ENSG00000256185                              Novel Transcript   
11935            RIEG2                             Rieger Syndrome 2   
11936             HNP1                      Hypertensive Nephropathy   
11937     LOC109279247                    FGF21/FUT1 Promoter Region   

                 Category Uniprot ID  Gifts        GC Id  Relevance score  \
0          Protein Coding     P04637     66  GC17M007661       205.116104   
1          Protein Coding     P08581     67  GC07P116672       143.600006   
2          Protein Coding     P38398     63  GC17M043044       140.350800   
3          Protein Coding     Q13315     66  GC11P108223       128.671478   
4          Protein Coding     P52701     61  GC02P047695       120.493660   
...                   ...        ...    ...          ...              ...   
11933            RNA Gene        NaN      9  GC15M074411         0.177408   
11934            RNA Gene        NaN      7  GC12M026335         0.177408   
11935       Genetic Locus        NaN      6  GC13U990027         0.177408   
11936       Genetic Locus        NaN      2  GC09U900671         0.177408   
11937  Functional Element        NaN      2  GC19P133376         0.177408   

                                          GeneCards Link  
0      https://www.genecards.org/cgi-bin/carddisp.pl?...  
1      https://www.genecards.org/cgi-bin/carddisp.pl?...  
2      https://www.genecards.org/cgi-bin/carddisp.pl?...  
3      https://www.genecards.org/cgi-bin/carddisp.pl?...  
4      https://www.genecards.org/cgi-bin/carddisp.pl?...  
...                                                  ...  
11933  https://www.genecards.org/cgi-bin/carddisp.pl?...  
11934  https://www.genecards.org/cgi-bin/carddisp.pl?...  
11935  https://www.genecards.org/cgi-bin/carddisp.pl?...  
11936  https://www.genecards.org/cgi-bin/carddisp.pl?...  
11937  https://www.genecards.org/cgi-bin/carddisp.pl?...  

[11938 rows x 8 columns]

In [23]:
# แสดงเฉพาะคอลัมน์ Gene
GeneCard_HCC1 = GeneCard_HCC['Gene Symbol']

# เปลี่ยนชื่อคอลัมน์ให้เหมือนกันเพื่อความสะดวกในการเปรียบเทียบ
GeneCard_HCC1 = GeneCard_HCC1.rename('Symbol')

# รวมตารางเข้าด้วยกันโดยใช้คอลัมน์ 'Symbol' เป็นตัวเชื่อม
NCBI_GeneCard_HCC = pd.merge(HCC1, GeneCard_HCC1, on='Symbol', how='inner')

# นับจำนวนแถวในตารางที่รวมกันแล้ว (ซึ่งก็คือจำนวนยีนที่ตรงกัน)
matched_genes_HCC = len(NCBI_GeneCard_HCC)

# แสดงผลลัพธ์
print(f"จำนวนยีน HCC ที่ตรงกันระหว่างข้อมูลจาก GeneCard และ NCBI คือ {matched_genes_HCC} ยีน")

# แสดงตารางที่รวมกันแล้ว
NCBI_GeneCard_HCC

จำนวนยีน HCC ที่ตรงกันระหว่างข้อมูลจาก GeneCard และ NCBI คือ 4347 ยีน


Symbol
0           TP53
1         CTNNB1
2           TPX2
3        ANGPTL8
4       HLA-DRB1
...          ...
4342   LINC01608
4343       HCG15
4344  ALKBH3-AS1
4345   PTOV1-AS1
4346       EGILA

[4347 rows x 1 columns]

### เปรียบเทียบยีน HCC 4347 ยีนที่ได้จากการ match ระหว่าง GeneCard และ NCBI กับ panel ของ Vizgen

In [25]:
# รวมตารางเข้าด้วยกันโดยใช้คอลัมน์ 'Symbol' เป็นตัวเชื่อม
HCC_vizgen = pd.merge(NCBI_GeneCard_HCC, panel1, on='Symbol', how='inner')

# นับจำนวนแถวในตารางที่รวมกันแล้ว (ซึ่งก็คือจำนวนยีนที่ตรงกัน)
HCC_vizgen_count = len(HCC_vizgen)

# แสดงผลลัพธ์
print(f"จำนวนยีน HCC ที่ match กันระหว่างข้อมูลจาก panel และ NCBI_GeneCard คือ: {HCC_vizgen_count} ตัว")

# แสดmerged_enstงตารางที่รวมกันแล้ว (ถ้าต้องการ)
HCC_vizgen

จำนวนยีน HCC ที่ match กันระหว่างข้อมูลจาก panel และ NCBI_GeneCard คือ: 318 ตัว


Symbol
0        TP53
1      CTNNB1
2    HLA-DRB1
3      PIK3CA
4         MET
..        ...
313      IDO2
314     PREX2
315     SMOC2
316  NDUFA4L2
317   CLEC14A

[318 rows x 1 columns]

In [26]:
# prompt: ลบข้อมูลที่ซ้ำในตาราง HCC_vizgen ออก

# Drop duplicate rows based on the 'Symbol' column
HCC_vizgen = HCC_vizgen.drop_duplicates(subset='Symbol')

# Display the DataFrame after removing duplicates
print(f"จำนวนยีนที่ตรงกันหลังจากลบข้อมูลซ้ำ: {len(HCC_vizgen)} ยีน")
HCC_vizgen

จำนวนยีนที่ตรงกันหลังจากลบข้อมูลซ้ำ: 318 ยีน


Symbol
0        TP53
1      CTNNB1
2    HLA-DRB1
3      PIK3CA
4         MET
..        ...
313      IDO2
314     PREX2
315     SMOC2
316  NDUFA4L2
317   CLEC14A

[318 rows x 1 columns]

In [29]:
# Specify the output file path in your Google Drive, including the file extension (.xlsx)
output_HCC = '/content/drive/MyDrive/Spatial/HCC_vizgen_file.xlsx'  # Changed this to include .xlsx

# Export the merged DataFrame to an Excel file
HCC_vizgen.to_excel(output_HCC, index=False)  # index=False prevents writing row indices to the file

print(f"Merged data exported to: {output_HCC}")

Merged data exported to: /content/drive/MyDrive/Spatial/HCC_vizgen_file.xlsx


# Xenium

## นำเข้าข้อมูล xenium

In [30]:
xenium = pd.read_csv('/content/drive/MyDrive/Spatial/Xenium_hSkin_gene_panel.csv')
xenium

Gene                  Annotation          based on paper  \
0       CD79A                     B cells                  B cell   
1       MEF2C                     B cells                     NaN   
2     PKHD1L1                     B cells                     NaN   
3        MZB1       B cells; Erythrocytes  Plasma B (plasma cell)   
4    TNFRSF17       B cells; Erythrocytes                     NaN   
..        ...                         ...                     ...   
255      RHOV         TSK_differentiating                     NaN   
256     IFI27  Vascular Endothelial Cells                     NaN   
257     SOX17  Vascular Endothelial Cells                     NaN   
258     SPRY1  Vascular Endothelial Cells                     NaN   
259   TSC22D1  Vascular Endothelial Cells                     NaN   

          Ensembl ID  
0    ENSG00000105369  
1    ENSG00000081189  
2    ENSG00000205038  
3    ENSG00000170476  
4    ENSG00000048462  
..               ...  
255  ENSG00000104140  
256  ENSG00000165949  
257  ENSG00000164736  
258  ENSG00000164056  
259  ENSG00000102804  

[260 rows x 4 columns]

In [32]:
# เปลี่ยนชื่อคอลัมน์ของ xenium จาก "Gene" เป็น "Symbol"
xenium = xenium.rename(columns={"Gene": "Symbol"})
xenium

Symbol                  Annotation          based on paper  \
0       CD79A                     B cells                  B cell   
1       MEF2C                     B cells                     NaN   
2     PKHD1L1                     B cells                     NaN   
3        MZB1       B cells; Erythrocytes  Plasma B (plasma cell)   
4    TNFRSF17       B cells; Erythrocytes                     NaN   
..        ...                         ...                     ...   
255      RHOV         TSK_differentiating                     NaN   
256     IFI27  Vascular Endothelial Cells                     NaN   
257     SOX17  Vascular Endothelial Cells                     NaN   
258     SPRY1  Vascular Endothelial Cells                     NaN   
259   TSC22D1  Vascular Endothelial Cells                     NaN   

          Ensembl ID  
0    ENSG00000105369  
1    ENSG00000081189  
2    ENSG00000205038  
3    ENSG00000170476  
4    ENSG00000048462  
..               ...  
255  ENSG00000104140  
256  ENSG00000165949  
257  ENSG00000164736  
258  ENSG00000164056  
259  ENSG00000102804  

[260 rows x 4 columns]

In [33]:
xenium1 = xenium['Symbol']
xenium1

0         CD79A
1         MEF2C
2       PKHD1L1
3          MZB1
4      TNFRSF17
         ...   
255        RHOV
256       IFI27
257       SOX17
258       SPRY1
259     TSC22D1
Name: Symbol, Length: 260, dtype: object

## หายีนที่เกี่ยวข้องกับ melanoma ที่เหมือนกันของ Vizgen กับ Xenium

In [31]:
melanoma_vizgen

Symbol
0    CDKN1A
1     CXCL1
2      BRAF
3       FAP
4      TP53
..      ...
294    IDO2
295   IL4I1
296   PREX2
297  CLEC4C
298   CCR10

[299 rows x 1 columns]

In [34]:
# รวมตารางเข้าด้วยกันโดยใช้คอลัมน์ 'Symbol' เป็นตัวเชื่อม
Melanoma_xeviz = pd.merge(melanoma_vizgen, xenium1, on='Symbol', how='inner')

# นับจำนวนแถวในตารางที่รวมกันแล้ว (ซึ่งก็คือจำนวนยีนที่ตรงกัน)
Melanoma_xeviz_count = len(Melanoma_xeviz)

# แสดงผลลัพธ์
print(f"จำนวนยีนที่เกี่ยวข้องกับ melanoma ที่ตรงกันระหว่างข้อมูลจาก vizgen  และ xenium คือ {Melanoma_xeviz_count} ยีน")

# แสดงตารางที่รวมกันแล้ว
Melanoma_xeviz

จำนวนยีนที่เกี่ยวข้องกับ melanoma ที่ตรงกันระหว่างข้อมูลจาก vizgen  และ xenium คือ 27 ยีน


Symbol
0      MYC
1    CXCR4
2     MMP2
3     IFNG
4     IL1B
5      KIT
6    CCND1
7   CXCL12
8    ITGB1
9    BIRC5
10    TP63
11   MKI67
12    IDO1
13     VWF
14  PDGFRA
15    IRF4
16    CCR7
17  CD40LG
18    ATF3
19    CDH5
20   ACTA2
21    CD8A
22   SFRP2
23   FBLN1
24    CD68
25    CD3D
26   IL4I1